packages/@lwc/ssr-compiler/src/compile-template/transformers/element.ts

```diff
@@ -246,6 +246,13 @@ export const Element: Transformer<IrElement | IrExternalComponent | IrSlot> = fu
         childContent = [];
     }
 
+    const isForeignElement = node.namespace !== HTML_NAMESPACE;
+    const hasChildren = childContent.length > 0;
+
+    if (isForeignElement && !hasChildren) {
+        return [bYield(b.literal(`<${node.name}`)), ...yieldAttrsAndProps, bYield(b.literal(`/>`))];
```

Comment:
This could be merged as a ternary in the return statement below.

Reply:
I also removed the separate return for void html elements and they return here too. They were also missing the scope class and failed the test once that was added.